client/components/LearnerViewDirective.js

```diff
@@ -465,8 +466,20 @@ tie.directive('learnerView', [function() {
         };
 
         $scope.navigateToQuestion = function(index) {
+          // Before the questionId is changed, save it for later use.
+          var currentQuestionId = 
+            $scope.questionIds[$scope.currentQuestionIndex];
           $scope.currentQuestionIndex = index;
           var questionId = $scope.questionIds[$scope.currentQuestionIndex];
+          // Before the current code is changed to starter code for 
+          // next question, save it
+          var currentQuestionCode = $scope.code;
+          // Store code, the order matters, need to save code before loading 
```

Comment:
// We need to save the code before loading so that the user will get
// their own code back if they click on the current question.